lib/ansible/module_utils/basic.py

```diff
@@ -624,6 +629,10 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
         if not self.no_log and self._verbosity >= 3:
             self._log_invocation()
 
+        if not self.no_log and self._debug and self._remote_log_target:
+            self.debug_logger = logging.getLogger(self.__class__.__name__)
```

Comment:
And for new style ziploader modules, a module scope 'log' object could work.

```
log = logging.getLogger(__name__)      # <-- does __name__ get weird on remote modules?
```

Then instead of ```self.debug_logger.debug(foo)```, it would be ```log.debug(foo)```

That would remove the need for the module introspect in AnsibleModule.log(). Then potentially of the rest of AnsibleModule.log() becomes a matter of which logging Handler is setup.